src/test/java/teammates/test/driver/TestProperties.java

```diff
@@ -86,20 +86,44 @@
 
             TEAMMATES_VERSION = extractVersionNumber(FileHelper.readFile("src/main/webapp/WEB-INF/appengine-web.xml"));
 
-            TEST_ADMIN_ACCOUNT = prop.getProperty("test.admin.account");
-            TEST_ADMIN_PASSWORD = prop.getProperty("test.admin.password");
+            if (isCiEnvironment() || isGodModeEnabled()) {
+                // For CI and GodMode, we do not read the account details from the test properties file, but generate
+                // random account names. This is for detection and prevention of hard-coded account names in test files.
 
-            TEST_INSTRUCTOR_ACCOUNT = prop.getProperty("test.instructor.account");
-            TEST_INSTRUCTOR_PASSWORD = prop.getProperty("test.instructor.password");
+                String dotSalt = "." + StringHelperExtension.generateSaltOfLength(8);
+                String dummyPassword = "anypassword";
```

Comment:
Alternatively you can just read the password fields as how they were before. It reduces the number of items in the if block.